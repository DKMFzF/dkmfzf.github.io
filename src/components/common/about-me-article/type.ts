import { ReactNode } from 'react';

export interface AboutMeArticleProps {
	children: ReactNode;
	subtitle?: string;
	className?: string;
	borderPrimaryColorFlag?: boolean;
	paddingFlag?: boolean;
	borderRadiusFlag?: boolean;
	borderFlag?: boolean;
}
