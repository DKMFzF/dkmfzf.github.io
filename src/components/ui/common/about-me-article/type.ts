import { ReactNode } from 'react';

export interface AboutMeArticleUIProps {
	children: ReactNode;
	subtitle?: string;
	className?: string;
	borderPrimaryColorFlag?: boolean;
	paddingFlag?: boolean;
	borderRadiusFlag?: boolean;
	borderFlag?: boolean;
}
