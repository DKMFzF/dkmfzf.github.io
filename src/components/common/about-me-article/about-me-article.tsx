import { FC } from 'react';
import { AboutMeArticleUI } from 'src/components/ui';
import { AboutMeArticleProps } from './type';

export const AboutMeArticle: FC<AboutMeArticleProps> = ({
	children,
	subtitle = '',
	className = '',
	borderPrimaryColorFlag = false,
	paddingFlag = false,
	borderRadiusFlag = false
}) => (
	<AboutMeArticleUI
		subtitle={subtitle}
		className={className}
		borderPrimaryColorFlag={borderPrimaryColorFlag}
		paddingFlag={paddingFlag}
		borderRadiusFlag={borderRadiusFlag}
	>
		{children}
	</AboutMeArticleUI>
);
