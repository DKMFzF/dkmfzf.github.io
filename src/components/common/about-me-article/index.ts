export { AboutMeArticle } from './about-me-article';
