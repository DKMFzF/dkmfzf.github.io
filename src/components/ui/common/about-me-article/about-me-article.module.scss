@use '~styles/variables' as *;

$padding-card-block: 40px;
$border-block: 1px solid $bg-primary-color;
$border-premary-card-color: 1px solid #cec2e6;

.aboutMe {
	&__block {
		align-content: center;
		border: $border-block;
	}

	&__subTitle {
		width: fit-content;
		font-size: 13px;
		text-transform: uppercase;
		opacity: $opacity;
	}

	&__paddingDefault {
		padding: $padding-card-block;
	}

	&__borderPrimaty {
		border: $border-premary-card-color;
	}

	&__borderRadiusDefault {
		border-radius: $section-border-radius;
	}
}
