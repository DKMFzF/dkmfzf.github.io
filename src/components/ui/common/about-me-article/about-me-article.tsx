import { FC } from 'react';
import styles from './about-me-article.module.scss';
import { AboutMeArticleUIProps } from './type';

export const AboutMeArticleUI: FC<AboutMeArticleUIProps> = ({
	children,
	subtitle,
	className = '',
	borderPrimaryColorFlag = false,
	paddingFlag = false,
	borderRadiusFlag = false,
	borderFlag = false
}) => (
	<article
		className={`${styles.aboutMe__block} ${className} 
		${borderPrimaryColorFlag ? styles.aboutMe__borderPrimaty : ''}
		${!paddingFlag ? styles.aboutMe__paddingDefault : ''}
		${borderRadiusFlag ? styles.aboutMe__borderRadiusDefault : ''}
		${borderFlag ? styles.aboutMe__borderFlag : ''}`}
	>
		{subtitle && (
			<header className={styles.aboutMe__subTitle}>
				<h3>{subtitle}</h3>
			</header>
		)}
		{children}
	</article>
);
