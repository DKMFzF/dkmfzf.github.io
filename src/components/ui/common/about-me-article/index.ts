export { AboutMeArticleUI } from './about-me-article';
