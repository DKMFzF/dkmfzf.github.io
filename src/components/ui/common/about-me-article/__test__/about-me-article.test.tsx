import { testRenderComponent } from '@utils-testing';
import { AboutMeArticleUI } from '../about-me-article';

describe('[AboutMeArticleUI]', () => {
	it('[All props true]: Renders about-me-article correctly and matches snapshot', () =>
		testRenderComponent(
			<AboutMeArticleUI
				subtitle='subtitle'
				className='classname'
				borderPrimaryColorFlag
				paddingFlag
				borderRadiusFlag
				borderFlag
				flexFlag
			>
				<p>Test element</p>
			</AboutMeArticleUI>
		));

	it('[All props false]: Renders about-me-article correctly and matches snapshot', () =>
		testRenderComponent(
			<AboutMeArticleUI>
				<p>Test element</p>
			</AboutMeArticleUI>
		));
});
