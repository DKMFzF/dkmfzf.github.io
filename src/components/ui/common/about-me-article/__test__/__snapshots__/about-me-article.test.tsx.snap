// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AboutMeArticleUI] [All props false]: Renders about-me-article correctly and matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="about-me__block  
		
		about-me__padding-default
		
		
		"
  >
    <p>
      Test element
    </p>
  </article>
</DocumentFragment>
`;

exports[`[AboutMeArticleUI] [All props true]: Renders about-me-article correctly and matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="about-me__block classname 
		about-me__border-primaty
		
		about-me__border-radius-default
		about-me__border-flag
		about-me__flex-flag"
  >
    <header
      class="about-me__sub-title"
    >
      <h3>
        subtitle
      </h3>
    </header>
    <p>
      Test element
    </p>
  </article>
</DocumentFragment>
`;
